### ISSUE TITLE
Prompts for background access, when whenInUse is granted

### ISSUE THREAD

**Issue by esbenvb** [transistorsoft/react-native-background-geolocation#1039]
We have made this app which is using this module for various purposes. Geofences, updating the location when using the app, and getting current location when needed. Everything works great. We have chosen only to prompt the user for access to location when the app is in use. This is an important thing for us, since we don't want the users to feel that we spy on them.

Even the geofences we have defined works great with the `WhenInUse` access granted - which is also what the docs of iOS/Android says.

BUT the app keeps prompting us for background location access. And I can't find a way to disable that. 

## Your Environment
* Plugin version: 3.6.2
* Platform: iOS 
* OS version: 13.4.1
* Device manufacturer / model: iPhone Xs Max 64 GB
* React Native version (`react-native -v`): 0.61.2
* Plugin config
```javascript <!-- Syntax-highlighting:  paste your code below -->
  handleOpenMapDirection = async () => {
    const { coordinate } = this.props
    const transportPlan = 'w'
    try {
      const position = await BackgroundGeolocation.getCurrentPosition({})
      OpenMapDirections(position.coords, coordinate, transportPlan)
    } catch (e) {
      console.warn(e)
    }
  }

```

## Expected Behavior
The app should not prompt the user as long as `WhenInUse` is enabled

## Actual Behavior
The app keeps asking the user for background location permission.

## Steps to Reproduce

1. Call `Permissions.request('location', { type: 'whenInUse' })`
2. Call `BackgroundGeolocation.getCurrentPosition({})`
3. Wait and it will show "Background location is not enabled"

## Context

![IMG_6157](https://user-images.githubusercontent.com/426051/80220211-cbe5fe80-8643-11ea-8821-308e61aeeb8d.PNG)

**Comment by christocracy** [comment]
See API docs `Config.locationAuthorizationRequest` and `Config.disableLocationAuthorizationAlert`.

**Comment by esbenvb** [comment]
Thank you for the tip, now at least I got rid of the prompt.

The thing is, that I would actually still like to warn the user if he has not authorized location tracking, so that's why I did not think of using `disableLocationAuthorizationAlert`.

But I also believe that if you configure this module with `WhenInUse` authorization, it should not complain if `always` is not enabled, but only complain if `WhenInUse` is not enabled.

That's why I filed this as bug
